java/src/com/google/idea/blaze/java/sync/projectstructure/JavaSourceFolderProvider.java

```diff
@@ -105,7 +105,7 @@ private static String derivePackagePrefix(File file, SourceFolder parentFolder)
     if (Strings.isNullOrEmpty(relativePath)) {
       return parentPackagePrefix;
     }
-    relativePath = relativePath.replaceAll(File.separator, ".");
+    relativePath = relativePath.replace(File.separator, ".");
```

Comment:
Thanks! We're already replacing File.separatorChar with '/' in the call to FileUtil#toCanonicalPath above, so the correct fix here should be:
`relativePath.replace('/', '.')`

I'll make the change upstream.